src/libraries/System.Private.CoreLib/src/System/Text/ASCIIUtility.cs

```diff
@@ -281,13 +327,20 @@ private static unsafe nuint GetIndexOfFirstNonAsciiByte_Sse2(byte* pBuffer, nuin
 
                 do
                 {
-                    Vector128<byte> firstVector = Sse2.LoadAlignedVector128(pBuffer);
-                    Vector128<byte> secondVector = Sse2.LoadAlignedVector128(pBuffer + SizeOfVector128);
+                    if (Sse2.IsSupported)
+                    {
+                        Vector128<byte> firstVector = Sse2.LoadAlignedVector128(pBuffer);
+                        Vector128<byte> secondVector = Sse2.LoadAlignedVector128(pBuffer + SizeOfVector128);
 
-                    currentMask = (uint)Sse2.MoveMask(firstVector);
-                    secondMask = (uint)Sse2.MoveMask(secondVector);
+                        currentMask = (uint)Sse2.MoveMask(firstVector);
+                        secondMask = (uint)Sse2.MoveMask(secondVector);
+                    }
+                    else
+                    {
+                        throw new PlatformNotSupportedException();
+                    }
 
-                    if ((currentMask | secondMask) != 0)
+                    if (ContainsNonAsciiByte_Sse2(currentMask) || ContainsNonAsciiByte_Sse2(secondMask))
```

Comment:
>ContainsNonAsciiByte_Sse2(currentMask) || ContainsNonAsciiByte_Sse2(secondMask) [](start = 24, length = 79)

`ContainsNonAsciiByte_Sse2(currentMask | secondMask)` to avoid an extra comparison.